sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -104,6 +119,14 @@ class InMemoryCatalog extends ExternalCatalog {
         }
       }
       // Remove the database.
+      val dbDefinition = catalog(db).db
+      try {
+        fs.delete(new Path(dbDefinition.locationUri), true)
```

Comment:
Just one possible issue. If we create a database with an non-existent directory, it could create `.../x/y/z/`, `.../x/y/` and `.../x/`. When we drop a database, we will just drop the leaf directory `.../x/y/z/`.